mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerLocalScan.java

```diff
@@ -253,6 +253,7 @@ public void testCreateTableWithColumnSpecification() throws IOException {
         testTables.locationForCreateTableSQL(identifier);
     runCreateAndReadTest(identifier, createSql, HiveIcebergStorageHandlerTestUtils.CUSTOMER_SCHEMA,
         PartitionSpec.unpartitioned(), data);
+    shell.executeStatement("DROP TABLE " + identifier);
```

Comment:
We already have a test case for `testCreateDropTable`, so I think this part is covered. Do I miss something?

Could we create specific test cases instead for dropping the Iceberg table backing Hive table and then dropping the Hive table? (should only run for non-HiveCatalog tests)

Reply:
> We already have a test case for testCreateDropTable, so I think this part is covered. Do I miss something? Maybe we should create one where we add some data as well, if it is relevant

That test case does not detect the difference of behaviour between Hive2/3 and Hive4 with regards to purge=TRUE, because if we create a table and not to do any additional snapshot-creating operations on the table, then `metadata.snapshots()` will give back an empty collection, hence the deleteFiles operation will not run during `dropTableData` (and you won't get FileNotFoundExc for Hive4). That's why I added the DROP TABLE to some tests where additional operations (like inserts) also take place.

On the other hand, not sure why `metadata.snapshots()` gives back an empty collection in this case.

> Could we create specific test cases instead for dropping the Iceberg table backing Hive table and then dropping the Hive table? (should only run for non-HiveCatalog tests)

Yes, will do.

Reply:
If it is not too inconvenient, create the snapshots directly on the iceberg tables, since it is faster than with hive inserts.

I think it is better to have a specific test case for both cases then

Reply:
Sure, I agree, will create two new tests for these specific scenarios